### ISSUE TITLE
Resolve minimum size for sequence-type schemas

### ISSUE THREAD

**Issue by ikitommi** [metosin/malli#363]
this is needed to figure out the minimum arity of function schemas with varargs. Could be a new Protocol like:

```clj
(defprotocol SequenceSchema
  (-min-size [this] "returns minimum size of thee sequence"))
```

implemented to `:tuple`, `:vector`, ... `:cat` ....

**EDIT** I think we can limit this to regex/sequence schemas. So:

```clj
(require '[malli.core :as m])

(m/minimum-size [:? int?]) ;=> 0
(m/minimum-size [:cat [:? string?] [:+ int?] [:repeat {:min 1} boolean?]) ;=> 2
(m/minimum-size [:alt [:+ int?] [:? boolean?]]) ;; => 0
```

**Comment by miikka** [comment]
A couple of examples would be nice to illustrate what is wanted here.

**Comment by ikitommi** [comment]
Added examples. Also, this is needed for #360.

**Comment by miikka** [comment]
Solved by #365.
